endpoints/openrtb2/auction_test.go

```diff
@@ -120,196 +191,178 @@ func TestExplicitUserId(t *testing.T) {
-			messageGetter: nilReturner,
-			expectedCode:  http.StatusOK,
-			aliased:       true,
-			accountReq:    true,
-		},
-		{
-			// Account is required, was provided in request and is found in the  blacklisted accounts map
-			dir:           "sample-requests/blacklisted",
-			file:          "blacklisted-acct.json",
-			payloadGetter: getRequestPayload,
-			messageGetter: getMessage,
-			expectedCode:  http.StatusServiceUnavailable,
-			accountReq:    true,
-		},
+func assertTestCaseData(t *testing.T, fileData []byte, testFile string) {
+	t.Helper()
+
+	// Retrieve values from JSON file
+	test := jsonParseTestFile(t, fileData, testFile)
+
+	test.Config = parseMaps(test.Config)
+
+	// Run test
+	actualCode, actualBidResponse := doRequest(t, *test)
+
+	// Assertions
+	assert.Equal(t, test.ExpectedReturnCode, actualCode, "Test failed. Filename: %s \n", testFile)
+
+	// Either assert bid response or expected error
+	if test.ExpectedReturnCode != 200 {
+		// Assert expected error
+		assert.True(t, strings.HasPrefix(actualBidResponse, test.ExpectedErrorMessage), "Test failed. %s. Filename: %s \n", actualBidResponse, testFile)
+	} else {
+		// Assert expected response
+		diffJson(t, testFile, []byte(actualBidResponse), test.ExpectedBidResponse)
+	}
+}
+
+func parseMaps(tc *testConfigValues) *testConfigValues {
+	if len(tc.BlacklistedAccountArr) > 0 {
+		tc.blacklistedAccountMap = make(map[string]bool, len(tc.BlacklistedAccountArr))
+		for _, account := range tc.BlacklistedAccountArr {
+			tc.blacklistedAccountMap[account] = true
+		}
 	}
-	for _, test := range tests {
-		test.assert(t)
+	if len(tc.BlacklistedAppArr) > 0 {
+		tc.blacklistedAppMap = make(map[string]bool, len(tc.BlacklistedAppArr))
+		for _, app := range tc.BlacklistedAppArr {
+			tc.blacklistedAppMap[app] = true
+		}
 	}
+	if len(tc.AdapterList) > 0 {
+		tc.adaptersConfig = make(map[string]config.Adapter, len(tc.AdapterList))
+		for _, adapterName := range tc.AdapterList {
+			tc.adaptersConfig[adapterName] = config.Adapter{Disabled: true}
+		}
+	}
+	return tc
 }
 
-// assertResponseFromDirectory makes sure that the payload from each file in dir gets the expected response status code
-// from the /openrtb2/auction endpoint.
-func (gr *getResponseFromDirectory) assert(t *testing.T) {
-	//t *testing.T, dir string, payloadGetter func(*testing.T, []byte) []byte, messageGetter func(*testing.T, []byte) []byte, expectedCode int, aliased bool) {
+// diffJson compares two JSON byte arrays for structural equality. It will produce an error if either
+// byte array is not actually JSON.
+func diffJson(t *testing.T, description string, actual []byte, expected []byte) {
 	t.Helper()
-	var filesToAssert []string
-	if gr.file == "" {
-		// Append every file found in `gr.dir` to the `filesToAssert` array and test them all
-		for _, fileInfo := range fetchFiles(t, gr.dir) {
-			filesToAssert = append(filesToAssert, gr.dir+"/"+fileInfo.Name())
+	diff, err := gojsondiff.New().Compare(actual, expected)
```

Comment:
This is different than we normally use to compare json, but it doesn't strike me as odd. If this looks good to others, I have no issue.